source/extensions/tracers/opentelemetry/samplers/dynatrace/sampler_config_provider.cc

```diff
@@ -8,10 +10,98 @@ namespace Extensions {
 namespace Tracers {
 namespace OpenTelemetry {
 
+static constexpr std::chrono::seconds INITIAL_TIMER_DURATION{10};
+static constexpr std::chrono::minutes TIMER_INTERVAL{5};
+
+namespace {
+
+bool reEnableTimer(Http::Code response_code) {
+  switch (response_code) {
+  case Http::Code::OK:
+  case Http::Code::TooManyRequests:
+  case Http::Code::InternalServerError:
+  case Http::Code::BadGateway:
+  case Http::Code::ServiceUnavailable:
+  case Http::Code::GatewayTimeout:
+    return true;
+  default:
+    return false;
+  }
+}
+
+std::chrono::milliseconds getTimeout(envoy::config::core::v3::HttpUri& http_uri) {
+  auto timeout =
+      std::chrono::milliseconds(DurationUtil::durationToMilliseconds(http_uri.timeout()));
+  // we want to ensure that we don't have more than one pending request
+  if (timeout > (TIMER_INTERVAL / 2)) {
+    timeout = (TIMER_INTERVAL / 2);
+  }
+  return timeout;
+}
```

Comment:
does this necessary? Seems you will only re-enable the timer when the `onSucess` or `onFailure` is called. So, that means you will only emit a new request after last request is completed?

Reply:
You are right, it is not possible to have more than one panding request. I have removed the adjustment in 535d99b7932c49f082bf5e8e1fda969708de1d32